### PR TITLE
Develop upstream sync 180806

### PR DESCRIPTION
**Pull request by sunway513**
Weekly upstream merge.

### CODE REVIEW

**Comment by deven-amd** [comment]
one of the failures `//tensorflow/python/keras:topology_test` seems to be due to the MIOpen HOME not set bug
one other failure `//tensorflow/python/kernel_tests:matrix_exponential_op_test` could potentially be solved by the alternate fix for HIP PR 457

may make sense to rerun CI on this PR, after PR 105 is merged.

**Comment by whchung** [comment]
jenkins: retest this please

File: configure.py [hunk 1]
```diff
@@ -1517,7 +1517,7 @@ def main():
     else:
       set_trisycl_include_dir(environ_cp)
 
-  set_action_env_var(environ_cp, 'TF_NEED_ROCM', 'ROCm', False)
```
Comment (whchung) [inline configure.py]: since we have `configure_input` now, can we leave this value default to `False`?

File: build [hunk 2]
```diff
@@ -8,6 +8,7 @@
 #
 # press enter all the way
 #
+PYTHON_BIN_PATH=/usr/bin/python cat ./configure_rocm | ./configure
```
Comment (whchung) [inline build]: this is incorrect. should be:

```
cat ./configure_rocm | PYTHON_BIN_PATH=/usr/bin/python ./configure
```

File: build_python3 [hunk 3]
```diff
@@ -8,6 +8,7 @@
 #
 # press enter all the way
 #
+PYTHON_BIN_PATH=/usr/bin/python3 cat ./configure_rocm | ./configure
```
Comment (whchung) [inline build_python3]: this is incorrect. should be:

```
cat ./configure_rocm | PYTHON_BIN_PATH=/usr/bin/python3 ./configure
```

File: configure.py (hunk 1 above)
Comment (sunway513) [reply configure.py]: yes, will update

File: build_python3 (hunk 3 above)
Comment (sunway513) [reply build_python3]: fixed in 42c8c83

File: build (hunk 2 above)
Comment (sunway513) [reply build]: fixed in 42c8c83

**Comment by whchung** [comment]
@deven-amd just want to understand how should we cope with this MIOpen related errors in certain test cases?

```
MIOpen Error: boost::filesystem::create_directories: Permission denied: "/.config"
MIOpen Error: boost::filesystem::create_directories: Permission denied: "/.cache/miopen"
```

**Comment by deven-amd** [comment]
@whchung you should not be seeing those....I updated the MIOpen fork (that the `Dockerfile.rocm` points to) to use the `$PWD` as the `$HOME` instead of the empty string. So in theory you should not paths which seem like $HOME is empty string.  This is puzzling.

**Comment by deven-amd** [comment]
@whchung , where did you see that error?

**Comment by deven-amd** [comment]
just looked at the log for build 275, and I see those errors...they are for the 11 tests that are affected by the MIOpen HOME bug...but for some reason HOME is being set to the empty string and not pwd.  cannot explain what is going on, will need to look into the container for the CI build

it should be building MIOpen with this comit

https://github.com/deven-amd/MIOpen/commit/6a96d439c695e08137d798b6e42b8c712c801189

**Comment by whchung** [comment]
for record keeping purpose:

http://205.234.28.200:21096/job/tensorflow-upstream-unit-tests/275

**Comment by whchung** [comment]
jenkins: retest this please

**Comment by deven-amd** [comment]
jenkins: retest this please

**Comment by deven-amd** [comment]
just for reference.

There were two MIOPen builds for PR 105 (which updates Dockerfile.rocm to build MIOpen)
1. build 269, `use empty string when HOME not set`, Failed, `rack-17` node
2. build 271, `use PWD shen HOME not set`, Passed, `master` node

Because the `RUN` lines (for building MIOpen) in the `Dockerfile.rocm` have not changed since build 269, docker always uses cache for those RUN steps for all subsequent builds. 

build 271 worked by luck because it got kicked off on a separate machine then build 269 (`master` as opposed to `rack-17`) and hence docker cache had no entry for RUN steps that build MIOpen, and redid those steps picking up the newer commit that uses PWD. 

builds 275 and 277 (the two build for this PR that are failing even even after the merge for PR 105) are both on `rack-17` node, and hence picking up the MIOpen build from cache (which uses empty string when HOME not set)

@parallelo was kind enough to clear out the docker cache on `rack-17` node, and if my theory holds the next build (278) should not use the cache for the MIOpen build and subsequently pass

**Comment by whchung** [comment]
more weirdness:

```
[91mCMake Error: Problem with archive_read_open_file(): Unrecognized archive format
CMake Error: Problem extracting tar: /tmp/cget-20-58-13-FJJw5-1/download/bzip2-1.0.6.tar.gz
[0m[91mCMake Error at /usr/local/share/cmake/cmakeget/CMakeGet.cmake:142 (message):
```

**Comment by whchung** [comment]
jenkins: retest this please

**Comment by whchung** [comment]
2 new failing tests:
```
//tensorflow/python/keras:topology_test
//tensorflow/python/kernel_tests:matrix_exponential_op_test
```

**Comment by whchung** [comment]
@sunway513 looking at the latest build log (280), it seems:


- `//tensorflow/python/keras:topology_test` : failing on known MIOpen issue. since MIOpen build is broken due to another unrelated issue to bzip2 let's add this to the whitelist for now.
- `//tensorflow/python/kernel_tests:matrix_exponential_op_test` : failing on known rocblas limitation. let's add this to the whitelist.

**Comment by sunway513** [comment]
@whchung done

**Comment by sunway513** [comment]
@whchung done pushing tags

**Pull request merged**